src/m365/spo/commands/homesite/homesite-remove.ts

```diff
@@ -54,23 +47,41 @@ class SpoHomeSiteRemoveCommand extends SpoCommand {
         const spoAdminUrl = await spo.getSpoAdminUrl(logger, this.debug);
         const reqDigest = await spo.getRequestDigest(spoAdminUrl);
 
-        const requestOptions: CliRequestOptions = {
-          url: `${spoAdminUrl}/_vti_bin/client.svc/ProcessQuery`,
-          headers: {
-            'X-RequestDigest': reqDigest.FormDigestValue
-          },
-          data: `<Request AddExpandoFieldTypeSuffix="true" SchemaVersion="15.0.0.0" LibraryVersion="16.0.0.0" ApplicationName="${config.applicationName}" xmlns="http://schemas.microsoft.com/sharepoint/clientquery/2009"><Actions><ObjectPath Id="28" ObjectPathId="27" /><Method Name="RemoveSPHSite" Id="29" ObjectPathId="27" /></Actions><ObjectPaths><Constructor Id="27" TypeId="{268004ae-ef6b-4e9b-8425-127220d84719}" /></ObjectPaths></Request>`
-        };
-
-        const res = await request.post<string>(requestOptions);
-
-        const json: ClientSvcResponse = JSON.parse(res);
-        const response: ClientSvcResponseContents = json[0];
-        if (response.ErrorInfo) {
-          throw response.ErrorInfo.ErrorMessage;
+        if (args.options.url) {
+          await this.removeHomeSiteByUrl(args.options.url, spoAdminUrl, logger);
         }
         else {
-          await logger.log(json[json.length - 1]);
+          await this.showDeprecationWarning(
+            logger,
+            commands.HOMESITE_REMOVE,
+            `${commands.HOMESITE_REMOVE} --url <url>`
+          );
```

Comment:
This works, but I would make it slightly clearer by not including the command in the warning.

Reply:
What I meant was, instead of using `this.showdeprecationwarning`, we should use `this.warn` and print a message that the current way the command works will change in the next major version where `url` will be required.